src/components/ImportRecord/ImportRecord.js

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage, injectIntl } from 'react-intl';
+import ky from 'ky';
+import { stripesConnect, withOkapiKy, CalloutContext } from '@folio/stripes/core';
+import { LoadingView } from '@folio/stripes/components';
+
+
+class ImportRecord extends React.Component {
+  static propTypes = {
+    id: PropTypes.string,
+    stripes: PropTypes.shape({
+      logger: PropTypes.object.isRequired,
+    }).isRequired,
+    resources: PropTypes.shape({
+      query: PropTypes.shape({
+        xid: PropTypes.string,
+      }).isRequired,
+    }).isRequired,
+    mutator: PropTypes.shape({
+      query: PropTypes.shape({
+        update: PropTypes.func.isRequired,
+      }).isRequired,
+    }).isRequired,
+    okapiKy: PropTypes.func.isRequired,
+    intl: PropTypes.shape({
+      formatMessage: PropTypes.func.isRequired,
+    }).isRequired,
+  };
+
+  static contextType = CalloutContext;
+
+  static manifest = Object.freeze({
+    query: {},
+  });
+
+  constructor(props) {
+    super(props);
+    this.loading = false;
+  }
+
+  componentDidUpdate() {
```

Comment:
It looks like you are running this only once. Could you move this to `componentDidMount`. After that we should be able to remove `this.loading`.

Reply:
This is resolved in 99ba2d27ea57207cdc4f52dbce7a9ce1873c6bca — which entails switching to use `withLocation` instead of he `query` resource, as the latter is not reliably present in `componentDidMount`.